modules/@apostrophecms/piece-type/ui/apos/components/AposPiecesManager.vue

```diff
@@ -276,23 +279,37 @@ export default {
         this.getPieces();
       }
     },
-    async edit(pieceId) {
-      const doc = await apos.modal.execute(this.options.components.insertModal, {
-        moduleName: this.moduleName,
-        docId: pieceId,
+    async edit(piece) {
+      let moduleName;
+      // Don't assume the piece has the type of the module,
+      // this could be a virtual piece type such as "submitted-draft"
+      // that manages docs of many types
+      if (piece) {
+        if (piece.slug.startsWith('/')) {
+          moduleName = '@apostrophecms/page';
```

Comment:
It seems this should now be `AposDocsManager.vue`.

Reply:
I don't think I understand what you're suggesting here. I'm determining the name of the module with the API suited to the document.

Reply:
The component is `AposPiecesManager`, but with this change it is no longer specific to pieces. I don't want that to be misleading or confusing.